docs/src/design.md

```diff
@@ -16,7 +16,51 @@ There are several functions in the interface for an [`AbstractAlgebraicAgent`](@
 
 These are collected into `ret`, which is an object that will be `true` if and only if all agents have returned `true`, and is otherwise the minimum of the numeric values (projection times) returned from each inner agent's step.
 
-![](assets/simloop.png)
+```mermaid
+flowchart TD
+
+    Start((Enter Program))-->Project[Set t equal to minimum \n projected time]:::GreenNode
+
+    Project-->PreWalk[Prestep inner agents]:::GreenNode
+
+    PreWalk -->|_prestep!| Inners([<:AbstractAlgebraicAgent]):::RedNode
+    
+    PreWalk --> Step[Step inner agents]:::GreenNode
+
+    Step -->|step!| Inners
+
+    subgraph inners
+    Inners
+    end
+
+    Ret([ret]):::RedNode
+
+    Inners -.-> Ret
+
+    Step --> LocalDecision{projected time == t}:::YellowNode
```

Comment:
I'd suggest something more explicit, e.g., `local projected time == t\n equals the min projected time`. What do you think?

Reply:
Yes, I agree that sounds much better.